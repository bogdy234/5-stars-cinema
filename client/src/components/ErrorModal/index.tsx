import React, { FC, ReactElement } from "react";
import CONSTANTS from "../../constants";
import Icon from "../Icon";
import Button from "../Button";
import Modal from "../Modal";

interface SuccessModalProps {
    showModal: boolean;
    onClose: () => void;
}

const { ERROR_MODAL } = CONSTANTS.TEXT;

const ErrorModal: FC<SuccessModalProps> = ({
    showModal,
    onClose,
}: SuccessModalProps): ReactElement => {
    return (
        <Modal showModal={showModal}>
            <div className={`flex justify-center items-center h-full`}>
                <div
                    className={`bg-white w-5/6 md:w-1/2 lg:w-1/3 h-1/2 rounded-3xl`}
                >
                    <div
                        className={`w-full h-1/2 bg-red-500 rounded-t-3xl flex justify-center items-center`}
                    >
                        <div className={"w-24 h-24"}>
                            <Icon
                                src={"/circle-xmark-solid.svg"}
                                alt={"success-icon"}
                            />
                        </div>
                    </div>
                    <div
                        className={`flex justify-center items-center flex-col`}
                    >
                        <p className={`pt-6 pb-6 text-2xl w-5/6 text-center`}>
                            {ERROR_MODAL.MESSAGE}
                        </p>
                        <Button
                            onClick={onClose}
                            text={ERROR_MODAL.CLOSE}
                            className={`bg-red-500 rounded-2xl w-40 h-16`}
                        />
                    </div>
                </div>
            </div>
        </Modal>
    );
};

export default ErrorModal;
