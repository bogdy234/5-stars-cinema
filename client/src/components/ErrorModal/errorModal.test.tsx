import renderer from "react-test-renderer";
import ErrorModal from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(<ErrorModal showModal onClose={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
