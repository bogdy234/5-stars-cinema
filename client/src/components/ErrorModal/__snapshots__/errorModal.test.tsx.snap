// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fixed z-10 left-0 top-0 w-full h-full overflow-auto bg-black bg-opacity-50 false"
>
  <div
    className="flex justify-center items-center h-full"
  >
    <div
      className="bg-white w-5/6 md:w-1/2 lg:w-1/3 h-1/2 rounded-3xl"
    >
      <div
        className="w-full h-1/2 bg-red-500 rounded-t-3xl flex justify-center items-center"
      >
        <div
          className="w-24 h-24"
        >
          <div
            className="undefined"
          >
            <img
              alt="success-icon"
              className="w-full h-full undefined "
              src="/circle-xmark-solid.svg"
            />
          </div>
        </div>
      </div>
      <div
        className="flex justify-center items-center flex-col"
      >
        <p
          className="pt-6 pb-6 text-2xl w-5/6 text-center"
        >
          A aparut o eroare. Te rog incearca mai tarziu!
        </p>
        <button
          className="relative bg-red-500 rounded-2xl w-40 h-16 undefined"
          onClick={[MockFunction]}
          type="button"
        >
          Inchide
        </button>
      </div>
    </div>
  </div>
</div>
`;
